modules/intensity_transform/src/intensity_transform.cpp

```diff
@@ -0,0 +1,76 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+using namespace cv;
+using namespace std;
+
+namespace cv {
+namespace intensity_transform {
+
+void logTransform(const Mat input, Mat& output)
+{
+    double maxVal;
+    minMaxLoc(input, NULL, &maxVal, NULL, NULL);
+    const double c = 255 / log(1 + maxVal);
+
+    if (input.channels() == 1)
+    {
+        output = input + 1;
+    }
+    else
+    {
+        output = input + Scalar(1, 1, 1);
+    }
+
+    output.convertTo(output, CV_64F);
+    cv::log(output, output);
+    output *= c;
+    output.convertTo(output, CV_8UC3);
```

Comment:
`Mat::convertTo()` with `alpha/beta` parameters can reduce number of operations:

```
Mat in_plus_1_64f;
input.convertTo(in_plus_1_64f, CV_64F, 1, 1.0f);  // input + 1
Mat log_64f;
log(in_plus_1_64f, log_64f);
log_64f.convertTo(output, CV_8UC3, c, 0.0f);  // log_64f * c
```

Reply:
That's useful! Thank you for the feedback!